2012-07-16  Alan Modra  <amodra@gmail.com>

	* config/rs6000/rs6000.md (*movdi_internal32): Use m constraint instead
	of o<>.

--- gcc/config/rs6000/rs6000.md	2012-07-16 10:14:43.198085150 +0200
+++ gcc/config/rs6000/rs6000.md	2012-07-16 10:33:31.856023210 +0200
@@ -10145,7 +10145,7 @@
 ; List r->r after r->"o<>", otherwise reload will try to reload a
 ; non-offsettable address by using r->r which won't make progress.
 (define_insn "*movdi_internal32"
-  [(set (match_operand:DI 0 "rs6000_nonimmediate_operand" "=o<>,r,r,*d,*d,m,r,?wa")
+  [(set (match_operand:DI 0 "rs6000_nonimmediate_operand" "=m,r,r,*d,*d,m,r,?wa")
 	(match_operand:DI 1 "input_operand" "r,r,m,d,m,d,IJKnGHF,O"))]
   "! TARGET_POWERPC64
    && (gpc_reg_operand (operands[0], DImode)
